### PR TITLE
portlist: remove NewPoller constructor

### LINKED ISSUE #8171: portlist: allow to receive updates for loopback addresses
After an internal discussion, there are use cases (such as using `tailscale serve`) where loopback addresses are of interest from within the portlist package. 

Ideally we can use a param to switch the behavior on off based on the caller.

### PR DESCRIPTION
**Pull request by marwan-at-work**
This is a follow up on PR #8172 and a breaking change that removes NewPoller. The issue with the previous PR was that NewPoller immediately initializes the underlying os implementation and therefore setting IncludeLocalhost as an exported field happened too late and cannot happen early enough. Using the zero value of Poller was also not an option from outside of the package because we need to set initial private fields

Fixes #8171

### CODE REVIEW

File: portlist/poller.go [hunk 1]
```diff
@@ -31,14 +31,15 @@ type Poller struct {
 
 	c chan List // unbuffered
 
+	initOnce sync.Once // guards init of private fields
```
Comment (bradfitz) [inline portlist/poller.go:34]: move this up a couple lines before `c`.  (normally mutexes and sync.once etc come before the stuff they guard)

File: portlist/poller.go [hunk 2]
```diff
@@ -71,54 +72,44 @@ var newOSImpl func(includeLocalhost bool) osImpl
...
+// all private fields. All exported methods must call this in a
+// Once, otherwise they may panic.
+func (p *Poller) init() {
+	if debugDisablePortlist() {
+		return
+	}
 	if newOSImpl != nil {
 		p.os = newOSImpl(p.IncludeLocalhost)
 	}
+	p.closeCtx, p.closeCtxCancel = context.WithCancel(context.Background())
```
Comment (bradfitz) [inline portlist/poller.go:91]: minor, but you don't want to call WithCancel if it'll never be called. Imagine you're LocalBackend and Check returns an error and you then give up on it.

So Check should probably call Close on error

Comment (marwan-at-work) [reply portlist/poller.go:90]: @bradfitz done

**Comment by marwan-at-work** [comment]
Closing in favor of https://github.com/tailscale/tailscale/pull/8275

**Pull request closed without merging**